runtime/src/bank.rs

```diff
@@ -2198,6 +2201,27 @@ impl Bank {
         });
     }
 
+    pub fn update_last_restart_slot(&self) {
+        let feature_flag = self
+            .feature_set
+            .is_active(&feature_set::last_restart_slot_sysvar::id());
+
+        if feature_flag {
+            let last_restart_slot = {
+                let tmp = self.hard_forks();
+                let hard_forks = tmp.read().unwrap();
+                hard_forks.iter().last().map(|(slot, _)| *slot).unwrap_or(0)
```

Comment:
I have a slight preference to something like this over using a variable named `tmp`
```suggestion
                let hard_forks = self.hard_forks();
                let hard_forks_r = hard_forks.read().unwrap();
                hard_forks_r.iter().last().map(|(slot, _)| *slot).unwrap_or(0)
```